### PR TITLE
New NP model, restructure theory uncertainties

### PR DESCRIPTION
**Pull request by kdlong**
- Update wremnants-data with new theory corrections with NP model
- Make theory corrections combine a helper class, to avoid passing the same info all the time
- Add some helper tools in card_tool, such as the ability to define process groups, e.g. "signal_samples"

### CODE REVIEW

**Comment by kdlong** [comment]
Low stats comparison:

This PR: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/PR262/2023_08_07/
Reference: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/ReferenceRuns/2023_08_07_bed7aab/

High stats comparison in progress.

This PR: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/PRlambdaNP/2023_08_07/
Reference: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/ScheduledBuilds/2023_08_05_bed7aab/

File: wremnants/CardTool.py [hunk 1]
```diff
@@ -51,15 +51,47 @@ def __init__(self, cardName="card.txt"):
...
+            processes = [processes]
+
+        expanded_procs = []
+        for proc in processes:
+            expanded_procs += self.expandProcess(proc)
+
+        return expanded_procs
+
+    def expandProcess(self, process):
+        if process in self.procGroups:
```
Comment (davidwalter2) [inline wremnants/CardTool.py:91]: return self.procGroups.get(process, [process])

**Review by davidwalter2 — COMMENTED** [review]
Can you write a few words about why the central values have changed? What underlying feature has changed?

**Comment by kdlong** [comment]
> Can you write a few words about why the central values have changed? What underlying feature has changed?

It's because the NP model changed to allow the term that's exponentiated in bT space (qT Fourier transform) to be signed. In the previous model "Omega" could not be less than zero, and the data favoured small values. This made it quite challenging to build a nice uncertainty band, since the down variation was bounded by zero. In the new model, taking lambda to zero is sensible and then the variation can be symmetric around +/- delta for some "reasonable" value.

Some similar changes are also made for gamma_nu ~ sign(c_nu)*omega_nu^2. Whereas in the old central value I kind of just picked good values by eye, we now understand better that apart from very low (<1 GeV) qT values only gamma_nu is relevant, so a good way to build a central value is to take gamma_nu = 0 and the +/- delta again for this parameter. 

The trade off is the central value arguably getting a bit worse, but it massively improves the uncertainty templates and the postfit values shouldn't care that much. Plus the data MC really isn't that bad.

**Comment by cippy** [comment]
Looks fine to me

**Pull request merged**